csharp/extractor/Semmle.Extraction.CSharp/Populators/DirectiveVisitor.cs

```diff
@@ -91,8 +99,10 @@ public IfDirectiveStackElement(Entities.IfDirective entity)
 
         public override void VisitIfDirectiveTrivia(IfDirectiveTriviaSyntax node)
         {
-            var ifStart = new Entities.IfDirective(cx, node);
+            var ifStart = Entities.IfDirective.Create(cx, node);
             ifStarts.Push(new IfDirectiveStackElement(ifStart));
+            if (node.IsActive && node.ConditionValue)
+                activeConditions.Add(ifStart);
```

Comment:
I think `IsActive && ConditionValue` is the same as `BranchTaken` for `if` and `elif`. We're actually collecting which are the conditional branches that were compiled (taken), aren't we? And in that case I think we need to add this logic to `VisitElseDirectiveTrivia` too. In case of an `else`, we would anyway need to use `BranchTaken`, so I would use that in all three cases.